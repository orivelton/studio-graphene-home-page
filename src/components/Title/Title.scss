@import '../../assets/scss/helper.scss';

.title {
  color: $night-rider;
  font-size: 52px;
  max-width: 266px;
  display: flex;
  flex-direction: column;
  align-items: center;
  z-index: 1;
  position: relative;
  padding: 0 18px;
  &__top {
    font-size: 35px;
    line-height: 0.9;
    display: flex;
    flex-direction: column;
    text-align: center;
  }

  &__text {
    position: relative;
    width: 236px;
    display: flex;
    flex-direction: column;
    align-items: center;
    font-size: 62px;
    &::before {
      content: '';
      background: $primary;
      display: flex;
      width: 114%;
      height: 6px;
      position: absolute;
      bottom: 35px;
      z-index: -1;
      opacity: 1;
      left: -18px;
    }
  }

  &__text-bottom {
    font-size: 18px;
    background: $primary;
    padding: 8px 9px 0;
  }
}