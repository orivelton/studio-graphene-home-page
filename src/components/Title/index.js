import React from 'react';
import './Title.scss';

const Title = ({textTop, text, textBottom, lineWhite, colorSecondary}) => (
  <div className={colorSecondary ? 'title title--secondary uppercase': 'title uppercase'}>
    <h3 className={lineWhite ? 'title__top title__top--white': 'title__top'}>
      {textTop && textTop}
      {text && <span className={lineWhite ? 'title__text title__text--line-white ': 'title__text'}>{text}</span>}
    </h3>
    {textBottom && <h4 className="title__text-bottom">{textBottom}</h4>}
  </div>
);

export default Title;
