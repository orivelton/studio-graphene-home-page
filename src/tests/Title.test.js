import React from 'react';
import renderer from 'react-test-renderer';
import Title from '../components/Title';

test('renders correctly Title', () => {
  const tree = renderer.create(<Title />).toJSON();
  expect(tree).toMatchSnapshot();
});
