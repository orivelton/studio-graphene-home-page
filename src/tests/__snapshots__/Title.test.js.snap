// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Title 1`] = `
<div
  className="title uppercase"
>
  <h3
    className="title__top"
  />
</div>
`;
